tasks/libs/common/gomodules.py

```diff
@@ -0,0 +1,344 @@
+"""Provides functions to import / export go modules from / to yaml files."""
+
+from __future__ import annotations
+
+import os
+import subprocess
+import sys
+from collections.abc import Callable
+from dataclasses import dataclass
+from functools import lru_cache
+from pathlib import Path
+from typing import ClassVar
+
+import yaml
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Have you considered using [pydantic](https://docs.pydantic.dev/latest/) to define and validate the yaml?